src/layout/dom.ts

```diff
@@ -171,6 +173,26 @@ export function update(node: Node, parent: Node, data: NodeInfo, source: Source)
     }
 }
 
+export function sameorigin(node: Node): boolean {
+    let output = false;
+    if (node.nodeType === Node.ELEMENT_NODE && (node as HTMLElement).tagName === Constant.IFRAME_TAG) {
+        let frame = node as HTMLIFrameElement;
+        try {
+            let doc = frame.contentDocument;
+            if (doc) {
+                iframeMap.set(frame.contentDocument, frame);
+                output = true;
+            }
+        } catch { /* do nothing */ }
+    }
+    return output;
+}
+
+export function iframe(node: Node): HTMLIFrameElement {
+    let doc = node.nodeType === Node.DOCUMENT_NODE ? node as Document : null;
+    return doc && iframeMap.has(doc) ? iframeMap.get(doc) : null;
```

Comment:
>iframeMap [](start = 18, length = 9)

Minor suggestion: Maybe iframeMap -. iframeDoc2FrameMap to indicate what the mapping is from/to in the name?

Reply:
Added in the comments to clarify. Not changing names to stay consistent with rest of the naming.